src/main/java/io/knotx/commons/http/request/AllowedHeadersFilter.java

```diff
@@ -35,18 +37,19 @@ public static AllowedHeadersFilter create(List<Pattern> patterns) {
     return new AllowedHeadersFilter(patterns);
   }
 
-  private static AllowedHeadersFilter create(Set<String> regexes) {
-    regexes = Objects.isNull(regexes) ? Collections.emptySet() : regexes;
-    List<Pattern> patterns = regexes.stream()
-        .map(regex -> Pattern.compile(regex, Pattern.CASE_INSENSITIVE))
+  private static AllowedHeadersFilter create(Set<String> expressions) {
+    expressions = Objects.isNull(expressions) ? Collections.emptySet() : expressions;
+    List<Pattern> patterns = expressions.stream()
+        .map(expression -> expression.replace(WILDCARD, WILDCARD_REGEX))
+        .map(expression -> Pattern.compile(expression, Pattern.CASE_INSENSITIVE))
         .collect(Collectors.toList());
 
     return AllowedHeadersFilter.create(patterns);
   }
 
-  public static class CaseInsensitive {
-    public static AllowedHeadersFilter create(Set<String> regexes) {
-      return AllowedHeadersFilter.create(regexes);
+  public static class WildcardCaseInsensitive {
```

Comment:
Please rethink if we can modify regexes / wildcards more "gently" - maybe we can not replace `.*` with `..*`?

Reply:
I reverted this changes.

After thinking about it, there would be more comfortable to use regular expressions instead of wildcards (as "*").